api/bases/dataplane.openstack.org_openstackdataplanenodesets.yaml

```diff
@@ -199,6 +199,8 @@ spec:
                 - ctlplaneInterface
                 - deploymentSSHSecret
                 type: object
+              controlPlane:
+                type: string
```

Comment:
Just a thought.  Right now, the expectation is that there is only one control plane per namespace - and its not clear that this will change anytime soon.  For simplicity sake, does it make sense to just assume this is the case for now.  In that case, this code becomes even simpler.

We don't need to add a parameter for the controlPlane.  The logic would be:
-- look for a control plane
  -- if one exists, check tls settings for it
  -- if more than one exists, error out
  -- if none exists, just continue  (this is assuming that deploying a dataplane is possible and makes sense without 
     a control plane)

When we have a requirement for more than one control plane, we can add this parameter?

What do you think @slagle @gibizer @SeanMooney

Reply:
makeing crd changes after we release if they are not purely additive will be expensive

in general i think we should keep this simple and be explicit by pass it in by name

if its not present we can either error, look it up or assume we are deploying the dataplane by itself for testing and no tls will be used.

i would prefer to error and make it a required field.

you are correct wew are unlikely to have more the one instance of the conolplane in a given namespaces however i think we would be better served by not having complex logic to try and discover the control plane.
i would add it if for no other reason then we are being consistent with the pattern we have already established where depencies should be passed by name.
we avoid trying to look up things automaitcally.

Reply:
^^ @jpodivin @slagle  @gibizer  thoughts?  I don't have any real preference here.  Also, https://github.com/openstack-k8s-operators/openstack-operator/pull/760 landed recently.  Should we be specifying a control plane name or a label?

Reply:
> We don't need to add a parameter for the controlPlane. The logic would be: -- look for a control plane -- if one exists, check tls settings for it -- if more than one exists, error out -- if none exists, just continue (this is assuming that deploying a dataplane is possible and makes sense without a control plane)

That is how the check for OpenStackVersion is implemented, and I would be ok with the same for the ControlPlane.

Erroring when it doesn't exist (whether there is a field or not), adds another dependency on the ControlPlane itself, not just openstack-operator. Effectively, we'd need to add the `openstack_deploy` target as a dep for `dataplane_kuttl` in install_yamls.

Reply:
The field is a good idea, making the connection explicit. It also makes sense since in the future we are going to have even tighter coupling with control plane.

I'm a bit less convinced about making it required. But it is a workable approach.

As for the question of logic. There are couple of options. If we want too keep the coupling loose:

1. Is ControlPlane field set? Yes: Try to get it and use it. Error if it doesn't exist.
2. Is there a 1 control plane in namespace? Yes: use it. No: go 3
3. Is there more than one control plane? Yes: error. No: go 4
4. Proceed without control plane.

If we want it slightly tighter:
1. Is ControlPlane field set? Yes: Try to get it and use it. Error if it doesn't exist.
2. Proceed without control plane.

And finally, the simplest option is to just make the field required and error out if control plane it points to doesn't exist.

We can, hypothetically, proceed from option 1 to 3 with minor modifications of the code.
But all of them require the controlPlane field to be present.